BootloaderCorePkg/Tools/PrepareBuildComponentBin.py

```diff
@@ -246,7 +257,15 @@ def Main():
         CopyBins (fsp_repo_dir, sbl_dir, fsp_inf)
 
     microcode_inf = os.path.join(sbl_dir, 'Silicon', silicon_pkg_name, 'Microcode', 'Microcode.inf')
-    CopyBins (ucode_repo_dir, sbl_dir, microcode_inf)
+    microcode_bins = CopyBins (ucode_repo_dir, sbl_dir, microcode_inf)
+
+    # Add microcode padding
+    if microcode_bins != None and len(sys.argv) > 4:
+        slot_size = int(sys.argv)
+        for microcode_bin in microcode_bins:
+            microcode_bin = os.path.join(sbl_dir,microcode_bin[1])
+            DoAddPaddingData (microcode_bin, slot_size)
```

Comment:
Wouldn't this sort of padding occur as part of the call to align_pad_file() within create_bootloader_image() of BuildLoader.py? This code here only pads each *.mcb file before they get combined into a *.bin file. That final *.bin file also gets padded.

Reply:
Existing padding has no relation with SLOT_SIZE. Since this patch is to add SLOT concept, I think it should work regarding UserExtensions section exists or not in the INF file.

Reply:
Should now just pad all *.mcb files, regardless of the UserExtensions section.